src/parsec.rs

```diff
@@ -168,6 +172,17 @@ impl ParsecMap {
 
         parsec.has_unpolled_observations()
     }
+
+    fn remove_old(&mut self) {
+        let mut iter = 0;
+        while self.map.len() > MAX_PARSECS && iter < 100 {
```

Comment:
Do not need the while, only ever add one at a time.
Should be a bit simpler than this really since the key are ordered.

Reply:
Yes we _should_ delete them at the same rate as they are created, this is mostly defense programming